### ISSUE TITLE
IntelliSense and mypy config support

### ISSUE THREAD

**Issue by athyuttamre** [microsoft/vscode-mypy#37]
Hello — I found this project while looking for mypy support in VS Code. I noticed Pylance uses Pyright for IntelliSense tips (on hover, etc.) but we use mypy in our project.

Is Intellisense support already built? When I installed the pre-release version, I did not see any type hints show on hover.

Similarly, is there support for reading mypy config from a pyproject.toml or setup.cfg file, or does this extension only support .ini files?

Thanks for working on this!

**Comment by karthiknadig** [comment]
@athyuttamre Sorry for the delayed response on this.

> Is Intellisense support already built? 

No, we are currently working on switching this to use `dmypy` and with that we should be able to implement the Type Analysis part. But, this would be a feature request.

> is there support for reading mypy config from a pyproject.toml or setup.cfg file, or does this extension only support .ini files?

We don't specifically choose INI files. All we do is launch `mypy` as it would be from a terminal, so it should detect settings as it normally should. You can also pass the config file via `"mypy-type-checker.args": ["--config-file", "pyproject.toml"]`, if for some reason it is not recognizing.

**Comment by vidhanio** [comment]
@elahd why not just set it as your default globally in the user `settings.json`?

**Comment by g0di** [comment]
> It looks like `dmypy` doesn't search for a config file if other arguments are passed in. The extension's default argument of `--follow-imports=skip` is causing trouble:
> 
> https://github.com/microsoft/vscode-mypy/blob/ea18cf0f25b53ae2f95b842806c23fb071adfa1c/package.json#L63-L65
> 
> When I manually set `mypy-type-checker.args` to `[]`, the extension picks up my `pyproject.toml` config without issue.
> 
> Would you consider removing this default?

I've give it a try and even without specifying the `--config-file=pyproject.toml` mypy server is actually reading properly my configuration. Note that I'm using my venv mypy which version is 1.4.1.
>this can be done through the following setting: `"mypy-type-checker.importStrategy": "fromEnvironment"`

However, I'm more concerned about the actual extension defaults `--follow-imports=skip` because this basically cause all imported modules in your source code to be cast as `Any` which lead to additional error/warnings when you actually use elements of those modules in your code.

For example, the following code 
```python
from pydantic import AnyHttpUrl, BaseSettings, Field


class Settings(BaseSettings):
     pass

```

Produce the following error:  `error: Class cannot subclass "BaseSettings" (has type "Any")  [misc]` 
>Basically because using `skip` for `follow-imports` turns the pydantic imports into `Any`

Is there is any particular reason to set that default mypy setting?

**Comment by karthiknadig** [comment]
> Is there is any particular reason to set that default mypy setting?

This was the default in the python extension, I used the same here. We can change this to better reflect what the community wants. 

There is a PR from community contributor updating this: https://github.com/microsoft/vscode-mypy/pull/76

**Comment by Avasam** [comment]
> It looks like `dmypy` doesn't search for a config file if other arguments are passed in. [...]

That sounds... unexpected, I also don't see anything in https://mypy.readthedocs.io/en/stable/mypy_daemon.html or https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-config-file suggesting this is intended behaviour. Maybe it's accidentally parsing as `--config-file=` or something as soon as there's arguments? It'd be worth raising an issue over at mypy.
